### ISSUE TITLE
[QST] Is make_empty_column() doing the right thing? 

### ISSUE THREAD

**Issue by nvdbaranec** [rapidsai/cudf#5508]
There's an interesting issue that comes up with make_empty_column() and LISTs.   Right now, if you do this

`make_empty_column(LIST)`

you will get an empty column as expected, but importantly it will have no children.  Since list columns rely on the child columns to store actual values (ints, floats, strings, or even other further lists) this means you end up losing all type information about the column.  So if I did this:

```
list_column_wrapper<int> a{{}};
list_column_wrapper<int> b{{}};
list_column_wrapper<int> c{{}};
D = concatenate(a, b, c);
```

The resulting column D would have no children and therefore no leaf column of type INT.  It would just have 3 rows, each of which is an empty list column of.....no type whatsoever.

This may be ok, but I feel like we're going to run into situations where this might become a problem.  

Fixing this is tricky because there's two fundamental ways we make empty columns.

```
empty_like(column_view)
make_empty_column(type)
```
empty_like() would be easy to make work, since you have all the information you need to propagate your empty leaf column.  But make_empty_column(LIST) is a problem.  You have no way of knowing "hey, what should my child type be?".

**Comment by jrhemstad** [comment]
I figured that we would need to add children to the `data_type` class for things like this. If not for lists, then probably for structs.

**Comment by harrism** [comment]
@nvdbaranec should this be turned into a BUG/FEA?

**Comment by nvdbaranec** [comment]
After discussions about this, preservation of hierarchies in output is what we decided to go with - reserving the option to add nesting to the `data_type` if needed down the road.

PR
https://github.com/rapidsai/cudf/pull/5816

**Comment by kkraus14** [comment]
Fixed by #5816
